vllm/model_executor/model_loader/loader.py

```diff
@@ -285,16 +285,17 @@ def _get_weights_iterator(
         tensorizer_args = self.tensorizer_config._construct_tensorizer_args()
         return tensorizer_weights_iterator(tensorizer_args)
 
-    def _load_model_unserialized(
+    def _load_model_serialized_cpu(
             self, model_config: ModelConfig, device_config: DeviceConfig,
             lora_config: Optional[LoRAConfig],
             vision_language_config: Optional[VisionLanguageConfig]
     ) -> nn.Module:
-        """Load an unserialized model with tensorizer.
+        """Load a serialized model with tensorizer to the CPU.
```

Comment:
Why only CPU? Don't we have `with torch.device(device_config.device):`?

Reply:
If the model is not vLLM-tensorized, trying to load the model to any other device destination other than CPU is not currently supported, even if `device_config.device` isn't for CPU. `TensorDeserializer` accepts a `device` parameter, and loads the model directly to that device. If loaded directly to the GPU, vLLM's loading architecture then needs to make the model a "vLLM model" (by instantiating its bespoke model layers for the supported loaded model and copying the weights to those instantiated layers on `device_config.device`). Tensorizer loads directly to the target device before it allows loaded weights to be `yield`ed, so if the target device is to the GPU, passing them off to vLLM's loading architecture causes a double load. I tried to circumvent this in my original PR by doing in-place transfers of weights from loaded to instantiated models rather than `torch.copy_`ing, but this isn't possible with vLLM's weight concatenation layers, which require copying.

I circumvented this by directly allowing for serializing ready-to-serve vLLM models, and loading non-vLLM models (like a HF `PreTrainedModel`) to the CPU to allow for vLLM to instantiate it and load it to the GPU without double-copying.

Reply:
I see - this makes sense, one thing I'm worried about is under certain scenario this will cause unexpected CPU RAM OOM, but I would imagine that as a rare case.

I do think we should add some documentation for `Tensorizer` in our main documentation page around its usage and on a high level how it works.